contractcourt/chain_watcher.go

```diff
@@ -292,11 +290,23 @@ func newChainWatcher(cfg chainWatcherConfig) (*chainWatcher, error) {
 		stateHintObfuscator: stateHint,
 		quit:                make(chan struct{}),
 		clientSubscriptions: make(map[uint64]*ChainEventSubscription),
-		fundingPkScript:     fundingPkScript,
-		heightHint:          heightHint,
 		fundingSpendNtfn:    spendNtfn,
 	}
 
+	// If this is a pending taproot channel, we need to register for a
+	// confirmation notification of the funding tx.
```

Comment:
add a shorten explanation of the reasoning behind this as a comment from: https://github.com/lightningnetwork/lnd/pull/9277/files#r1864911793

Reply:
i think you are referencing the wrong place? this is about keysend cannot be recognized in `RegisterSpend`